users/views.py

```diff
@@ -68,11 +76,206 @@ def get_serializer_class(self):
 
 class UserViewSet(viewsets.ModelViewSet):
     queryset = CustomUser.objects.all()
-    permission_classes = (IsOwnerOrAdmin,)
+    permission_classes = (AuthenticatedAdminOrUser,)
 
     def get_serializer_class(self):
         if self.action == Action.RETRIEVE.value:
             return UserDetailSerializer
         if self.action == Action.UPDATE.value:
             return UserUpdateSerializer
         return UserSerializer
+
+
+@login_required
+def index(request):
+    # Number of visits to this view, as counted in the session variable.
+    num_visits = request.session.get('num_visits', 0)
+    request.session['num_visits'] = num_visits+1
+
+    # Render the HTML template index.html with the data in the context variable.
+    return render(
+        request,
+        'home.html',
+        context={
+            'num_visits': num_visits},  # num_visits appended
+    )
+
+
+class SignUp(APIView):
+    serializer_class = CustomRegisterSerializer
+    renderer_classes = [renderers.TemplateHTMLRenderer]
+    template_name = 'signup.html'
+
+    @classmethod
+    def get(cls, request):
+        serializer = CustomRegisterSerializer(context={'request': request})
+        return Response({'serializer': serializer})
+
+    @classmethod
+    def post(cls, request):
+        serializer = CustomRegisterSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response({
+                'serializer': serializer,
+                'errors': serializer.errors,
+            })
+        serializer.save(request)
+        return redirect('login')
+
+
+class UserCreate(APIView):
+    renderer_classes = [renderers.TemplateHTMLRenderer]
+    template_name = 'user_create.html'
+
+    @classmethod
+    def get(cls, request):
+        serializer = UserCreateSerializer(context={'request': request})
+        return Response({'serializer': serializer})
+
+    @classmethod
+    def post(cls, request):
+        serializer = UserCreateSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response({
+                            'serializer': serializer,
+                            'errors': serializer.errors,
+        })
+        email = serializer.validated_data.get('email')
+        serializer.save()
+        user = CustomUser.objects.get(email=email)
+        user.set_password('passwduser')
+        user.save()
+        return redirect('custom-users-list')
+
+
+class UserUpdate(APIView):
+    renderer_classes = [renderers.TemplateHTMLRenderer]
+    template_name = 'user_update.html'
+
+    @classmethod
+    def get(cls, request, pk):
+        user = get_object_or_404(CustomUser, pk=pk)
+        if request.user.user_type == "ADMIN":
+            serializer = UserDetailSerializer(user, context={'request': request})
+        else:
+            serializer = UserUpdateSerializer(user, context={'request': request})
```

Comment:
These two serializers names needs to be changed because they are misleading. Both of them are used to updating users details, but one of them is just used to updating user by admin. Please try change names for clear ones and do it in separate function to have in `get()` method just clear names what exactly this whole block of code does.

Reply:
Done